src/drivers/intel/dmic.c

```diff
@@ -1609,79 +1581,81 @@ static int dmic_probe(struct dai *dai)
 	}
 	dai_set_drvdata(dai, dmic);
 
+	dmic->private = rzalloc(SOF_MEM_ZONE_RUNTIME_SHARED, 0, SOF_MEM_CAPS_RAM,
+				sizeof(*dmic->private));
```

Comment:
aren't we back to where we started with this? Before all these fields were embedded in `struct dmic_pdata` now they are in a separate structure, which forces an additional allocation and this rather large diff. I understand the part with the `global` pointer, that now uses `platform_shared_get()` instead of the current cache-uncache translation for each use - this is an improvement! But the rest - why do we need it?

Reply:
yep, lets keep it a simple change.

Reply:
I think this is now done - hope so!